### PR TITLE
feat: migrate to new JSX transform

### PR DESCRIPTION
**Pull request by u4aew**
### Description 
Added `react-jsx` to **tsconfig.json** | **tsconfig.test.json**, updated rules in **.eslintrc**, updated presets in **.babelrc**  and removed `React` imports.

[docs](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html)

issue: #2003

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -32,7 +28,8 @@
         "import/consistent-type-specifier-style": ["error", "prefer-top-level"],
         "@typescript-eslint/prefer-ts-expect-error": "error",
         "@typescript-eslint/consistent-type-imports": ["error", {"prefer":  "type-imports", "fixStyle": "separate-type-imports"}],
-        "complexity": "off"
+        "complexity": "off",
+        "react/react-in-jsx-scope": "off"
```
Comment (ogonkov) [inline .eslintrc:32]: Guide also suggest `react/jsx-uses-react`

File: .eslintrc [hunk 2]
```diff
@@ -14,10 +14,6 @@
         ],
         "no-restricted-syntax": [
             "error",
-            {
-                "selector": "ImportDeclaration[source.value='react'] :matches(ImportNamespaceSpecifier, ImportSpecifier)",
-                "message": "Please use import React from 'react' instead."
-            },
```
Comment (amje) [inline .eslintrc:20]: Using `React` namespace is our codestyle. So `import * as React from 'react'` should be the only option.
To accomplish this, change the rule to:
```json
{
    "selector": "ImportDeclaration[source.value='react'] :matches(ImportDefaultSpecifier, ImportSpecifier)",
    "message": "Please use `import * as React from 'react'` instead."
}
```

File: .babelrc [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
-  "presets": [
-    [
-      "@babel/preset-env",
-      {
-        "targets": {
-          "chrome": 100
-        }
-      }
+    "sourceType": "unambiguous",
+    "presets": [
```
Comment (amje) [inline .babelrc:3]: Please keep current preset order

File: .eslintrc (hunk 1 above)
Comment (amje) [reply .eslintrc:33]: We can simply extend `plugin:react/jsx-runtime` which does the job. No need to off the rules here.

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Comment by amje** [comment]
@u4aew Thanks for contribution!

**Pull request merged**